pom.xml

```diff
@@ -400,11 +400,38 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-resources-plugin</artifactId>
+        <executions>
+          <!-- prepare site content by filtering ${project.*} values-->
+          <execution>
+            <id>filter-site</id>
+            <phase>pre-site</phase>
+            <goals>
+              <goal>copy-resources</goal>
+            </goals>
+            <configuration>
+              <outputDirectory>${project.build.directory}/site-src</outputDirectory>
+              <resources>
+                <resource>
+                  <directory>src/site</directory>
+                  <filtering>true</filtering>
+                </resource>
+              </resources>
+            </configuration>
+          </execution>
+        </executions>
+        <configuration>
+          <escapeString>\</escapeString>
+        </configuration>
+      </plugin>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-site-plugin</artifactId>
         <configuration>
           <locales>en,es,ja,fr,zh_CN,ko</locales>
+          <siteDirectory>${project.build.directory}/site-src</siteDirectory>
```

Comment:
Hi @hazendaz ,
Do those changes in pom.xml look OK to you?
It seems that the goal is to activate the variable substitution, but I'm not sure how or if it affects our already complicated site generation process.

If you are not sure, I'll revert the change.
Variable substitutions are nice, but not crucial.

Reply:
FYI, I copy it from https://github.com/mybatis/spring-boot-starter/blob/master/pom.xml

Reply:
Yeah, but you also added some extra config like `escapeString`.

This review process is consuming my time a lot, so I would appreciate if you could do these extra stuff in separate PRs.

Reply:
note: I love spring boot, just don't love that they leave maven users hanging by practically living off mavens defunct super pom and many developers don't understand that enough and wonder why builds are not stable as they could be.  When you see this at scale it becomes very clearly evident since spring failed in that regard, its always better to use their second option in dependency management which makes the choices they did make mute.

Reply:
and then...not use their choices at all unless that came from maven or well documented why.

Reply:
@hazendaz Sorry, I just want to use a value substitution instead of `x.x.x`. I didn't get your point. I just use **mybatis**-springboot-starter's workaround. Now, follow @harawata I revert it.

Reply:
Thank you for the update, @awxiaoxian2020 !
I'll review as soon as I find the time.

@hazendaz 
Thank you for the comment!
I'm not sure, but could it be related to https://github.com/mybatis/spring/pull/763 ?
Anyway, I'll forget about it for now :P